src/main/java/seedu/address/logic/commands/group/AddToGroupCommand.java

```diff
@@ -0,0 +1,131 @@
+        this.groupName = groupName;
+        this.members = members;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getPersonList();
+        Group existingGroup;
+        try {
+            existingGroup = model.getGroup(groupName);
+        } catch (GroupNotFoundException gnfe) {
+            throw new CommandException(String.format(MESSAGE_GROUP_NOT_EXISTS, groupName));
+        }
+        Group newGroup = this.createGroupWithAddedMembers(existingGroup, members, lastShownList);
+        model.setGroup(existingGroup, newGroup);
+        try {
+            model.setGroup(existingGroup, newGroup);
+        } catch (DuplicateGroupException e) {
+            throw new CommandException(e.getMessage());
+        }
+        return new CommandResult(formatSuccessCommandResultMessage());
+    }
+
+    /**
+     * Formats the message to output to the user after the execution of the command.
+     * This is non-trivial as new entries and duplicates are taken into account to
+     * output the correct string format.
+     *
+     * @return a string which should go into the CommandResult.
+     */
+    private String formatSuccessCommandResultMessage() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(String.format(MESSAGE_ADD_TO_GROUP_SUCCESS, groupName));
+        this.newMembers.forEach(person -> builder.append(person.getName()).append('\n'));
+        if (!this.duplicates.isEmpty()) {
+            builder.append(String.format(MESSAGE_ADD_TO_GROUP_DUPLICATES, groupName));
+            this.duplicates.forEach(person -> builder.append(person.getName()).append('\n'));
+        }
+        return builder.toString();
+    }
+
+    /**
+     * To enforce immutability, this function creates a new group with the added members.
+     *
+     * @param existingGroup The group object of the group before addition of members.
+     * @param members The indices of the members to be added, based on the lastShownList.
+     * @param lastShownList The list of Person's objects last displayed to the user.
+     * @return a new Group object with the added person objects.
+     */
+    private Group createGroupWithAddedMembers(Group existingGroup,
+                                                     List<Index> members, List<Person> lastShownList) {
+        Group newGroup = new Group(this.groupName);
+        newGroup.setPersons(existingGroup.asUnmodifiableObservableList());
+        members.stream()
+                .map(member -> lastShownList.get(member.getZeroBased()))
+                .forEach(person -> {
+                    try {
+                        newGroup.add(person);
+                        newMembers.add(person);
+                    } catch (DuplicatePersonException dpe) {
+                        this.duplicates.add(person);
+                    }
+                });
```

Comment:
Maybe something for next time but should `Group` have a copy constructor where we can do 
``` Java
Group existingGroup = new Group("abc");
Group copyGroup = new Group(existingGroup); // copies the existing Group
```

Reply:
Yep. Let me open an issue for this in 1.5.